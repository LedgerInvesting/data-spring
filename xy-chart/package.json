{
  "name": "@data-spring/xy-chart",
  "version": "0.9.6",
  "main": "build/index.js",
  "scripts": {
    "build": "webpack"
  },
  "description": "Components and utility functions for using react-spring animations in data-ui/xy-chart data series visualizations https://ledgerinvesting.github.io/data-spring/",
  "repository": "https://github.com/LedgerInvesting/data-spring.git",
  "keywords": [
    "d3",
    "react",
    "chart",
    "visualization",
    "plot",
    "motion",
    "animated",
    "animation",
    "spring",
    "data-ui",
    "react-spring"
  ],
  "author": "Matt Liu <matt@ledgerinvesting.com>",
  "license": "MIT",
  "dependencies": {
    "@babel/runtime": "^7.0.0"
  },
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-loader": "^7.1.5",
    "babel-eslint": "^8.2.2",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "@data-ui/xy-chart": "0.0.54",
    "react-spring": "^7.2.8",
    "react": "^16.0.0",
    "react-dom": "^16.0.0",
    "eslint": "^4.19.1",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-react": "^7.7.0",
    "webpack": "^4.20.2",
    "webpack-cli": "^3.1.1"
  }
}
