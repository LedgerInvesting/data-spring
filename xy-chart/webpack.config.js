var path = require('path');

module.exports = {
    mode: 'production',
    entry: './src/index.js',
    output: {
        path: path.resolve('build'),
        filename: 'index.js'
    },
    module: {
        rules: [
            {
                test: /\.jsx?$/,
                exclude: /(node_modules)/,
                use: 'babel-loader'
            }
        ]
    }
}
