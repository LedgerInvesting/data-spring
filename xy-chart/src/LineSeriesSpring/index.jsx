import React from 'react';
import { LineSeries } from '@data-ui/xy-chart';
import { Spring, config } from 'react-spring';
import { calcFlattenedStart, constructArray } from '../utils/motionUtil';


const LineSeriesSpring = (props) => {
  const motionPoints = calcFlattenedStart(props);

  return (
    <Spring
      {...props}
      config={props.springConfig}
      from={motionPoints.from}
      to={motionPoints.to}
    >
      {values => {
        return (
          <LineSeries
            {...props}
            data={constructArray(values, props.data)}
          />
        );
      }}
    </Spring>
  );
};

LineSeriesSpring.displayName = 'LineSeriesSpring';
LineSeriesSpring.defaultProps = {
  animationStart: 'bottomLeft',
  springConfig: config.default,
  customAnimationStart: (point) => point,
};
export default LineSeriesSpring;
