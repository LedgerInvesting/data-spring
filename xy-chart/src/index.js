import PointSeriesSpring from './PointSeriesSpring/index.jsx';
import LineSeriesSpring from './LineSeriesSpring/index.jsx';
import AreaSeriesSpring from './AreaSeriesSpring/index.jsx';

export {
  PointSeriesSpring as PointSeriesSpring,
  LineSeriesSpring as LineSeriesSpring,
  AreaSeriesSpring as AreaSeriesSpring,
};
