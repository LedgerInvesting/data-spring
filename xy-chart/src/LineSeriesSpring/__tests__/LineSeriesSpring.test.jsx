import React from 'react';
import LineSeriesSpring from '../';
import { config } from 'react-spring';
import { XYChart } from '@data-ui/xy-chart';
import renderer from 'react-test-renderer';


it('renders successfully', () => {
  const tree = renderer
    .create(
      <XYChart
        width={700}
        height={350}
        xScale={{ type: 'linear', domain: [0, chartBounds.xMax] }}
        yScale={{ type: 'linear', domain: [0, chartBounds.yMax] }}
      >
        <LineSeriesSpring
          strokeWidth={0.3}
          curve="monotoneX"
          fillOpacity=".75"
          fill="#000"
          stroke="#000"
          animationStart="bottom"
          data={testSeries.xySeries}
          bounds={chartBounds}
          springConfig={config.gentle}
        />
      </XYChart>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with default animationStart props', () => {
  const tree = renderer
    .create(
      <XYChart
        width={700}
        height={350}
        xScale={{ type: 'linear', domain: [0, chartBounds.xMax] }}
        yScale={{ type: 'linear', domain: [0, chartBounds.yMax] }}
      >
        <LineSeriesSpring
          strokeWidth={0.3}
          curve="monotoneX"
          fillOpacity=".75"
          fill="#000"
          stroke="#000"
          data={testSeries.xySeries}
          bounds={chartBounds}
          springConfig={config.gentle}
        />
      </XYChart>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with default springConfig props', () => {
  const tree = renderer
    .create(
      <XYChart
        width={700}
        height={350}
        xScale={{ type: 'linear', domain: [0, chartBounds.xMax] }}
        yScale={{ type: 'linear', domain: [0, chartBounds.yMax] }}
      >
        <LineSeriesSpring
          strokeWidth={0.3}
          curve="monotoneX"
          fillOpacity=".75"
          fill="#000"
          stroke="#000"
          animationStart="bottom"
          data={testSeries.xySeries}
          bounds={chartBounds}
        />
      </XYChart>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const testSeries = {
  xySeries: [
    {
      x: 1,
      y: 10,
    },
    {
      x: 1,
      y: 7,
    },
    {
      x: 3,
      y: 0,
    },
    {
      x: 10,
      y: 1,
    },
  ],
}

const chartBounds = {
  xMin: 0,
  yMin: 0,
  xMax: 9,
  yMax: 10,
};
