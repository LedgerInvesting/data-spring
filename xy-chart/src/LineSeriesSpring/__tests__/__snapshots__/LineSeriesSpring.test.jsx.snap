// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<svg
  height={350}
  role="img"
  width={700}
>
  <g
    className="cx-group"
    transform="translate(64, 64)"
  >
    <g>
      <path
        className="vx-linepath"
        d="M63.55555555555555,222C105.92592592592592,222,148.29629629629628,222,190.66666666666666,222C338.96296296296293,222,487.25925925925924,222,635.5555555555555,222"
        fill="none"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        stroke="#000"
        strokeDasharray={null}
        strokeDashoffset={0}
        strokeLinecap="round"
        strokeWidth={0.3}
        style={null}
      />
      <g
        className="vx-linepath-glyphs"
      >
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`renders with default animationStart props 1`] = `
<svg
  height={350}
  role="img"
  width={700}
>
  <g
    className="cx-group"
    transform="translate(64, 64)"
  >
    <g>
      <path
        className="vx-linepath"
        d="M0,222Z"
        fill="none"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        stroke="#000"
        strokeDasharray={null}
        strokeDashoffset={0}
        strokeLinecap="round"
        strokeWidth={0.3}
        style={null}
      />
      <g
        className="vx-linepath-glyphs"
      >
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`renders with default springConfig props 1`] = `
<svg
  height={350}
  role="img"
  width={700}
>
  <g
    className="cx-group"
    transform="translate(64, 64)"
  >
    <g>
      <path
        className="vx-linepath"
        d="M63.55555555555555,222C105.92592592592592,222,148.29629629629628,222,190.66666666666666,222C338.96296296296293,222,487.25925925925924,222,635.5555555555555,222"
        fill="none"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        stroke="#000"
        strokeDasharray={null}
        strokeDashoffset={0}
        strokeLinecap="round"
        strokeWidth={0.3}
        style={null}
      />
      <g
        className="vx-linepath-glyphs"
      >
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
        <a
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          role="presentation"
          xlinkHref="#"
        />
      </g>
    </g>
  </g>
</svg>
`;
